From a742d55ed36a0bd3b6346563665b951dad9ad469 Mon Sep 17 00:00:00 2001
From: root <root@fit-build-116.mtl.labs.mlnx>
Date: Tue, 5 Apr 2022 21:35:55 +0300
Subject: [PATH backport v6.1 2/3] mlxsw: minimal: Downstream: Ignore error
 reading SPAD register

SPAD register is not supported for IB systems.

Signed-off-by: Vadim Pasternak <vadimp@nvidia.com>
---
 drivers/net/ethernet/mellanox/mlxsw/minimal.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/mellanox/mlxsw/minimal.c b/drivers/net/ethernet/mellanox/mlxsw/minimal.c
index 15116d9305f8..7396b37bc880 100644
--- a/drivers/net/ethernet/mellanox/mlxsw/minimal.c
+++ b/drivers/net/ethernet/mellanox/mlxsw/minimal.c
@@ -53,6 +53,7 @@ struct mlxsw_m_port {
 
 static int mlxsw_m_base_mac_get(struct mlxsw_m *mlxsw_m)
 {
+#if 0
 	char spad_pl[MLXSW_REG_SPAD_LEN] = {0};
 	int err;
 
@@ -60,6 +61,7 @@ static int mlxsw_m_base_mac_get(struct mlxsw_m *mlxsw_m)
 	if (err)
 		return err;
 	mlxsw_reg_spad_base_mac_memcpy_from(spad_pl, mlxsw_m->base_mac);
+#endif
 	return 0;
 }
 
-- 
2.20.1

